flink-runtime/src/main/java/org/apache/flink/runtime/operators/sort/LargeRecordHandler.java

```diff
@@ -218,7 +221,7 @@ public long addRecord(T record) throws IOException {
 		return offset;
 	}
 	
-	public MutableObjectIterator<T> finishWriteAndSortKeys(List<MemorySegment> memory) throws IOException {
+	public MutableObjectIterator<T> finishWriteAndSortKeys(List<MemorySegment> memory) throws IOException, IllegalAccessException, TemplateException, InstantiationException, CompileException, InvocationTargetException, NoSuchMethodException, ClassNotFoundException {
```

Comment:
I'm thinking that we should encapsulate all these exceptions into one `SorterCodegenException` (or just `CodegenException`) and throw only that from the sorter factory method, to avoid having to declare this litany of exceptions (this same list occurs at multiple places).

Reply:
Or, maybe even better to catch all of these in the sorter factory method (`SorterFactory.createSorter`), and create a non-codegen sorter when any of these exceptions happen (and log a warning).